snuba/query/parser/expressions.py

```diff
@@ -159,19 +207,29 @@ def parse_aggregation(
 
     columns_expr = [parse_expression(column) for column in columns if column]
 
-    if isinstance(parsed_expression, str):
-        # Simple aggregation with snuba syntax ["count", ["c1", "c2"]]
-        return FunctionCall(alias, parsed_expression, tuple(columns_expr))
-    elif (
-        # Simple Clickhouse expression with no snuba syntax
-        # ["ifNull(count(somthing), something)", None, None]
-        isinstance(parsed_expression, (FunctionCall, CurriedFunctionCall))
-        and not columns_expr
-    ):
-        return replace(parsed_expression, alias=alias)
-    elif isinstance(parsed_expression, FunctionCall) and columns_expr:
-        # Mix of clickhouse syntax and snuba syntax that generates a CurriedFunction
-        # ["f(a)", "b", None]
-        return CurriedFunctionCall(alias, parsed_expression, tuple(columns_expr),)
+    matched = re.fullmatch("[a-zA-Z_][a-zA-Z0-9_]*", aggregation_function)
+
+    if not bool(matched):
```

Comment:
you can make this code more readable by inverting the condition (reduces the indentation).

```
if matched is not None:
   return FunctionCall(alias, aggregation_function, tuple(columns_expr))

elif ....
 ....
```